File: services/echo/internal/structures/disposal_claim.go
package structures

type DisposalClaim struct {
	Id         string     `json:"id" bson:"_id,omitempty"`
	UserId     string     `json:"user_id"`
	OperatorId string     `json:"operator_id"`
	Token      string     `json:"token"`
	Credits    float32    `json:"credits"`
	IsClaimed  bool       `json:"is_claimed"`
	Disposals  []Disposal `json:"disposals"`
	Weight     float32    `json:"weight"`

	/*
	   public double Credits => Disposals.Sum(x => x.Credits);
	   public float Weight => Disposals.Sum(x => x.Weight);
	*/
}

File: services/upx/Entities/DisposalClaim.cs
using Foundation.Common.Entities;
using Foundation.Common.Roles;

namespace Foundation.Services.UPx;

public class DisposalClaim : BaseEntity
{
    public string? UserId { get; set; } = default!;

    public required string OperatorId { get; set; }

    public required string Token { get; set; }

    public double Credits => Disposals.Sum(x => x.Credits);

    public required bool IsClaimed { get; set; }

    public float Weight => Disposals.Sum(x => x.Weight);

    public required Disposal[] Disposals { get; set; }
}
